### ISSUE TITLE
System.IO.DirectoryNotFoundException: Could not find a part of the path

### ISSUE THREAD

**Issue by basarat** [atom/atom#5459]
This may not be distinct from "path too long exception" (tracked in https://github.com/atom/atom/issues/5109). But "path too long" it is no where to be found in this error log (possibly same cause though).

```log
Program: Starting Squirrel Updater: --install .
Program: Starting install, writing to C:\Users\_actual_user_name_\AppData\Local\SquirrelTemp
CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.FileNotFoundException: Could not find file 'C:\Users\_actual_user_name_\AppData\Local\atom\packages\RELEASES'.
File name: 'C:\Users\_actual_user_name_\AppData\Local\atom\packages\RELEASES'
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__28.MoveNext()
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\packages
CheckForUpdateImpl: Reading RELEASES file from C:\Users\_actual_user_name_\AppData\Local\SquirrelTemp
CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
ApplyReleasesImpl: Found partially applied release folder, killing it: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\locales
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\apm
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\apm\bin
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\benchmark
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\dot-atom
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\exports
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\dot-atom\packages
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\keymaps
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\01ee27e5c6782c660d6a00ddfd7723d1db97a6c2
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\apm\lib
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\01ee27e5c6782c660d6a00ddfd7723d1db97a6c2\content
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\01ee27e5c6782c660d6a00ddfd7723d1db97a6c2\content\00d84ff0df63c8da5a07c6621a8eb7be4a1db577
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\01ee27e5c6782c660d6a00ddfd7723d1db97a6c2\content\05e3e02fbddfaabb3ab78bb7e1f5353b997cf77a
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\apm\native-module
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\menus
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\benchmark\fixtures
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content\00d84ff0df63c8da5a07c6621a8eb7be4a1db577
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\apm\templates
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\apm\templates\bundle
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content\05e3e02fbddfaabb3ab78bb7e1f5353b997cf77a
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content\0614d2216a194818b8f2c6eb75fc57c47b6eb2f9
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content\0e20080c4e715a935aa2dc2d3698fd4e60f04e54
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\01ee27e5c6782c660d6a00ddfd7723d1db97a6c2\content\0614d2216a194818b8f2c6eb75fc57c47b6eb2f9
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content\12aa1656c321c62369257935ee76935fdc335119
Utility: Starting to delete folder: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0\resources\app\less-compile-cache\1b4a81223fa1cca931f69003fa65cfaf1d05a72f\content\14e6be27d4665afa28c51dc26e6adf8ba1525579
............... MUCH MORE........................................
ApplyReleasesImpl: Writing files to app directory: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0
IEnableLogger: Failed to write file: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\_actual_user_name_\AppData\Local\Temp\nuget\ebhccch5.z4a\lib\net45\resources\app\node_modules\jasmine-tagged\node_modules\jasmine-focused\node_modules\jasmine-node\node_modules\coffeestack\node_modules\coffee-script\lib\coffee-script\browser.js'.
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
at NuGet.PhysicalPackageFile.GetStream()
at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
IEnableLogger: Failed to install package to app dir: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\_actual_user_name_\AppData\Local\Temp\nuget\ebhccch5.z4a\lib\net45\resources\app\node_modules\jasmine-tagged\node_modules\jasmine-focused\node_modules\jasmine-node\node_modules\coffeestack\node_modules\coffee-script\lib\coffee-script\browser.js'.
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
at NuGet.PhysicalPackageFile.GetStream()
at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
at Squirrel.UpdateManager.ApplyReleasesImpl.copyFileToLocation(FileSystemInfo target, IPackageFile x)
at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__80(IPackageFile x)
at System.Collections.Generic.List`1.ForEach(Action`1 action)
at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__7f()
at System.Threading.Tasks.Task.InnerInvoke()
at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Squirrel.UpdateManager.ApplyReleasesImpl.<installPackageToAppDir>d__86.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Squirrel.Utility.<LogIfThrows>d__4b`1.MoveNext()
```

**Comment by izuzak** [comment]
> ```
ApplyReleasesImpl: Writing files to app directory: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0
IEnableLogger: Failed to write file: C:\Users\_actual_user_name_\AppData\Local\atom\app-0.177.0: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\_actual_user_name_\AppData\Local\Temp\nuget\ebhccch5.z4a\lib\net45\resources\app\node_modules\jasmine-tagged\node_modules\jasmine-focused\node_modules\jasmine-node\node_modules\coffeestack\node_modules\coffee-script\lib\coffee-script\browser.js'.
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
```

cc @paulcbetts maybe?

**Comment by paulcbetts** [comment]
@basarat Does this happen always for you? Or just this once?

**Comment by basarat** [comment]
Always. On one particular computer. I have three other computers where it works fine.

**Comment by emarques** [comment]
I've downloaded Atom 0.178 setup for Windows and also got the same error

```github
Program: Starting Squirrel Updater: --install .
Program: Starting install, writing to C:\Users\emarques.QUIDGEST\AppData\Local\SquirrelTemp
CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\emarques.QUIDGEST\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__28.MoveNext()
CheckForUpdateImpl: Reading RELEASES file from C:\Users\emarques.QUIDGEST\AppData\Local\SquirrelTemp
CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
ApplyReleasesImpl: Writing files to app directory: C:\Users\emarques.QUIDGEST\AppData\Local\atom\app-0.178.0
IEnableLogger: Failed to write file: C:\Users\emarques.QUIDGEST\AppData\Local\atom\app-0.178.0: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass90.<copyFileToLocation>b__8f()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
IEnableLogger: Failed to install package to app dir: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass90.<copyFileToLocation>b__8f()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
   at Squirrel.UpdateManager.ApplyReleasesImpl.copyFileToLocation(FileSystemInfo target, IPackageFile x)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass86.<installPackageToAppDir>b__83(IPackageFile x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass86.<installPackageToAppDir>b__82()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<installPackageToAppDir>d__88.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__4b`1.MoveNext()
```

**Comment by basarat** [comment]
@emarques I didn't get `System.IO.PathTooLongException`. That would be https://github.com/atom/atom/issues/5109

**Comment by emarques** [comment]
Sorry, I had just read the first error, which was

```
System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\emarques.QUIDGEST\AppData\Local\atom\packages\RELEASES'.
``` 

And didn't check the rest of error log... 

I should get more coffee...

**Comment by basarat** [comment]
Fixed with v0.180.0
Closing for now.
